web/src/OrganizationEditPage.js

```diff
@@ -294,6 +364,38 @@ class OrganizationEditPage extends React.Component {
             />
           </Col>
         </Row>
+        <Row style={{marginTop: "20px"}} >
+          <Col style={{marginTop: "5px"}} span={(Setting.isMobile()) ? 22 : 2}>
+            {Setting.getLabel(i18next.t("general:Password obfuscator"), i18next.t("general:Password obfuscator - Tooltip"))} :
+          </Col>
+          <Col span={22} >
+            <Select virtual={false} style={{width: "100%"}}
+              value={this.state.organization.passwordObfuscatorType}
+              onChange={(value => {this.updatePasswordObfuscator("type", value);})}>
+              {
+                [
+                  {id: "plain", name: "Plain"},
+                  {id: "aes", name: "AES"},
+                  {id: "des", name: "DES"},
+                ].map((obfuscatorType, index) => <Option key={index} value={obfuscatorType.id}>{obfuscatorType.name}</Option>)
+              }
+            </Select>
+          </Col>
+        </Row>
+        {(this.state.organization.passwordObfuscatorType !== "" && this.state.organization.passwordObfuscatorType !== "plain") ? (
+          <Row style={{marginTop: "20px"}} >
+            <Col style={{marginTop: "5px"}} span={(Setting.isMobile()) ? 22 : 2}>
+              {Setting.getLabel(i18next.t("general:Password obf key"), i18next.t("general:Password obf key - Tooltip"))} :
+            </Col>
+            <Col span={22} >
+              <Input
+                value={this.state.organization.passwordObfuscatorKey}
+                onChange={(e) => {this.updatePasswordObfuscator("key", e.target.value);}}
+                status={(!this.state.passwordObfuscatorKeyValid && this.state.passwordObfuscatorKeyErrorMessage) ? "error" : undefined}
+              />
+              {!this.state.passwordObfuscatorKeyValid && this.state.passwordObfuscatorKeyErrorMessage && <Row style={{marginTop: "20px"}} > <div style={{color: "red", marginTop: "-20px"}}>{this.state.passwordObfuscatorKeyErrorMessage}</div> </Row>}
```

Comment:
imho, if possible, use the validation function of `Form.Item` will be better

Reply:
I tried to use Form.Item to require the key to match the regular expression. But I couldn't handle the problem of synchronously updating the rules and values ​​of Form.Item.
When I updated the rules and values ​​at the same time, it would report an error for my new key value based on the old rules.
I then tried to use setTimeout to update the value after the rules, but when I wanted to use update-organization to update the key that met the rules, it would report an error saying that I didn't meet the rules.
Before issuing the update-organization request, I used this.state.passwordObfuscatorKeyValid to indicate whether the key meets the requirements. This value will be updated every time the value of Form.Item is updated.

I put this part of the attempt in my own casdoor repository https://github.com/ZhaoYP-2001/casdoor/tree/passwdForm